views/common/detail_tabs/document.py

```diff
@@ -0,0 +1,583 @@
+)
+from admission.forms.admission.document import (
+    UploadFreeDocumentForm,
+    RequestFreeDocumentForm,
+    RequestDocumentForm,
+    RequestAllDocumentsForm,
+    ReplaceDocumentForm,
+    UploadDocumentForm,
+)
+from admission.infrastructure.utils import get_document_from_identifier
+from admission.mail_templates import (
+    ADMISSION_EMAIL_REQUEST_FAC_DOCUMENTS_GENERAL,
+    ADMISSION_EMAIL_REQUEST_FAC_DOCUMENTS_CONTINUING,
+    ADMISSION_EMAIL_REQUEST_SIC_DOCUMENTS_GENERAL,
+    ADMISSION_EMAIL_REQUEST_FAC_DOCUMENTS_DOCTORATE,
+    ADMISSION_EMAIL_REQUEST_SIC_DOCUMENTS_DOCTORATE,
+    ADMISSION_EMAIL_REQUEST_SIC_DOCUMENTS_CONTINUING,
+)
+from admission.templatetags.admission import CONTEXT_GENERAL, CONTEXT_DOCTORATE, CONTEXT_CONTINUING
+from admission.utils import format_academic_year
+from admission.views.doctorate.mixins import LoadDossierViewMixin, AdmissionFormMixin
+from base.models.entity_version import EntityVersion
+from base.models.person import Person
+from base.utils.htmx import HtmxPermissionRequiredMixin
+from infrastructure.messages_bus import message_bus_instance
+from osis_common.utils.htmx import HtmxMixin
+
+__namespace__ = 'document'
+
+__all__ = [
+    'DocumentView',
+    'UploadFreeCandidateDocumentView',
+    'UploadFreeInternalDocumentView',
+    'RequestFreeCandidateDocumentView',
+    'DocumentDetailView',
+    'RequestCandidateDocumentView',
+    'DeleteDocumentView',
+    'ReplaceDocumentView',
+    'UploadDocumentByManagerView',
+]
+
+
+class DocumentView(LoadDossierViewMixin, AdmissionFormMixin, HtmxPermissionRequiredMixin, HtmxMixin, FormView):
+    template_name = 'admission/document/base.html'
+    htmx_template_name = 'admission/document/request_all_documents.html'
+    permission_required = 'admission.view_documents_management'
+    urlpatterns = {'document': ''}
+    form_class = RequestAllDocumentsForm
+
+    retrieve_documents_command = {
+        CONTEXT_GENERAL: general_education_commands.RecupererDocumentsPropositionQuery,
+    }
+
+    @cached_property
+    def is_fac(self):
+        return ProgramManager.belong_to(self.request.user.person)
+
+    @cached_property
+    def deadline(self):
+        return datetime.date.today() + datetime.timedelta(days=15)
+
+    @cached_property
+    def documents(self) -> List[EmplacementDocumentDTO]:
+        return (
+            message_bus_instance.invoke(
+                self.retrieve_documents_command[self.current_context](
+                    uuid_proposition=self.admission_uuid,
+                )
+            )
+            if self.current_context in self.retrieve_documents_command
+            else []
+        )
+
+    @cached_property
+    def requestable_documents(self):
+        return (
+            [
+                document
+                for document in self.documents
+                if document.statut == StatutEmplacementDocument.A_RECLAMER.name and document.type in EMPLACEMENTS_FAC
+            ]
+            if self.is_fac
```

Comment:
Plutot mettre la condition dans la compréhension ? 

`if document.statut == StatutEmplacementDocument.A_RECLAMER.name and (self.is_fac and document.type in EMPLACEMENTS_FAC or not self.is_fac and document.type not in EMPLACEMENTS_FAC)` à simplifier si possible